src/Symfony/Component/Intl/Resources/emoji/build.php

```diff
@@ -130,7 +130,7 @@ public static function buildGitHubRules(array $emojisCodePoints): iterable
             $maps[$codePointsCount][$emoji] = ":$shortCode:";
         }
 
-        return ['github' => self::createRules($maps)];
+        return ['emoji-github' => self::createRules($maps), 'github-emoji' => array_flip(self::createRules($maps))];
```

Comment:
why creating rules twice here ?

Reply:
The existing code has already produced two files, "emoji-github.php" and "github-emoji", to facilitate conversion from emoji to Github code and vice versa

The deleted lines have been replaced with the new code:  

https://github.com/symfony/symfony/pull/49147/commits/ab7d00967b571eebf46fb198c540230602504147#diff-64df07e7ad990f3a461c8f1f4af4b7496fe2df6feb757214e78f27ee72db8eebL172

https://github.com/symfony/symfony/pull/49147/commits/ab7d00967b571eebf46fb198c540230602504147#diff-64df07e7ad990f3a461c8f1f4af4b7496fe2df6feb757214e78f27ee72db8eebL184

Reply:
Instead of calling `createRules` twice, you could create it once and reuse it to flip it.